### PR TITLE
[networking_mapping] Add instances mapping logic

### PR DESCRIPTION
**Pull request by pablintino**
As a pull request owner and reviewers, I checked that:
- [x] Appropriate testing is done and actually running
- [x] Appropriate documentation exists and/or is up-to-date:
  - [x] README in the role
  - [x] Content of the docs/source is reflecting the changes

### CODE REVIEW

**Comment by danpawlik** [comment]
recheck

File: plugins/module_utils/net_map/networking_env_definitions.py [hunk 1]
```diff
@@ -49,28 +49,46 @@ class MappedInstanceNetwork:
...
     ip_v6: typing.Optional[ipaddress.IPv6Interface] = None
-    hostname: typing.Optional[str] = None
     mtu: typing.Optional[int] = None
     parent_interface: typing.Optional[str] = None
     vlan_id: typing.Optional[int] = None
 
 
+@dataclasses.dataclass(frozen=True)
+class MappedInstance:
+    """Defines an instance and it's attached networks
```
Comment (rebtoor) [inline plugins/module_utils/net_map/networking_env_definitions.py:79]: `its` ?

**Review by rebtoor — COMMENTED** [review]
just a minor typo. 

What a great job you did here @pablintino :clap:

File: plugins/module_utils/net_map/networking_mapper.py [hunk 2]
```diff
@@ -84,6 +87,335 @@ def _map_host_network_range_to_output(
     )
 
 
+class NetworkingInstanceMapper:
+    """Converts the Networking Definition and facts into a MappedInstance
+
+    Handles the conversion of the Networking Definition into a
+    MappedInstance for a given instance.
+    The mapper uses the Networking Definition as the main source of true
```
Comment (raukadah) [inline plugins/module_utils/net_map/networking_mapper.py:95]:
```suggestion
    The mapper uses the Networking Definition as the main source of truth
```

**Review by raukadah — COMMENTED** [review]
/approve

**Review by raukadah — COMMENTED** [review]
/lgtm

**Review by rebtoor — APPROVED** [review]

**Pull request merged**